client/src/hooks/useCreateUser.js

```diff
@@ -18,22 +18,59 @@ export const useCreateUser = (
   initialRedirectUrl
 ) => {
   const [createUser, setCreateUser] = useLocalStorage('createUser', {})
+  const [orcidInfo, setOrcidInfo] = useLocalStorage('orcidInfo', {})
   const [currentStep, setCurrentStep] = useLocalStorage('currentStep', '')
   const [steps, setSteps] = useLocalStorage('steps', [])
+  const [needsEmail, setNeedsEmail] = useLocalStorage('needsEmail', false)
   const { user, setUser, setToken, setLoginRedirectUrl } = useUser()
 
+  const save = () => {
+    setCreateUser({ ...createUser })
+  }
+
   React.useEffect(() => {
-    // If ORCID has returned the auth code to this page, sign the user in
-    if (queryCode && !user) {
-      createAndLoginUser(queryCode, initialRedirectUrl)
+    let needsSave = false
+
+    if (email && !createUser.email) {
+      createUser.email = email
+      needsSave = true
+    }
+
+    if (grants && !createUser.grants) {
+      createUser.grants = grants
+      needsSave = true
+    }
+
+    if (needsSave) {
+      save()
```

Comment:
does this need to be in a useEffect and then set to false after?

Reply:
Looks like it does not need to be in there. I added the lines setting it to "false" as well.